### ISSUE TITLE
Migration from Nx 8 -> Nx 12 fails at update-10-0-0

### ISSUE THREAD

**Issue by hfournier** [nrwl/nx#7100]
## Current Behavior

The migration fails with:
```
Running migration update-10-0-0
Unregistered task "RunUpdate" in schematic "update-10-0-0".
C:\Users\henri\Documents\Develop\Websites\sta\node_modules\@nrwl\workspace\node_modules\yargs\build\lib\yargs.js:1132
                throw err;
                ^

Error: Command failed: C:\Users\henri\AppData\Local\Temp\tmp-35980-ulSv73rL5siN\node_modules\.bin\tao migrate --run-migrations
```
the step that fails is:
```
    {
      "version": "10.0.0-beta.1",
      "description": "Upgrades Angular and Angular CLI to 10.0.0",
      "factory": "./src/migrations/update-10-0-0/update-10-0-0",
      "package": "@nrwl/angular",
      "name": "update-10-0-0"
    },
```

## Expected Behavior

Successful migration

## Steps to Reproduce

Ran:
`npx @nrwl/tao@latest migrate latest`
`npm install`
`nx migrate --run-migrations`

### Environment

  Node : 14.17.0
  OS   : win32 x64
  npm  : 6.14.7

  nx : Not Found
  @nrwl/angular : 12.9.0
  @nrwl/cli : 12.9.0
  @nrwl/cypress : 12.9.0
  @nrwl/devkit : 12.9.0
  @nrwl/eslint-plugin-nx : Not Found
  @nrwl/express : Not Found
  @nrwl/jest : 12.9.0
  @nrwl/linter : 12.9.0
  @nrwl/nest : 12.9.0
  @nrwl/next : Not Found
  @nrwl/node : 12.9.0
  @nrwl/nx-cloud : Not Found
  @nrwl/react : Not Found
  @nrwl/schematics : Not Found
  @nrwl/tao : 12.9.0
  @nrwl/web : Not Found
  @nrwl/workspace : 12.9.0
  @nrwl/storybook : 12.9.0
  @nrwl/gatsby : Not Found
  typescript : 4.3.5

**Comment by hfournier** [comment]
I reset the folder back to before the migration, deleted node_modules, ran npm install, then tried the migration again, and got the same result. So I decided to bypass Nx and update manually. For anyone else running into a similar situation here are the steps I followed. The project was already on Angular 9, but Nx was on 8, so I ran:
```
ng update @nrwl/workspace@9.80
npm install
```
I attempted:
```
ng update @nrwl/workspace@10.4.15
```
but it failed trying to update to Angular 10. However, it blocked being able to use ng update. So I edited the `decorate-angular-cli.js` file and commented out the following code:
```
// if (process.argv[2] === 'update') {
//   const { output } = require('@nrwl/workspace');
//   output.error({
//     title: '"ng update" is deprecated in favor of "nx migrate". Read more: https://nx.dev/latest/angular/workspace/update'
//   });
//   throw new Error();
// }
```
when I attempted to update Angular CLI to 10, the update failed due to a TypeScript dependency conflict with NestJS 6. I wonder if this is why the migration failed on version 10?
So I updated NestJS with:
```
ng update @angular/cli@10.2.3
ng update @angular/material@10.2.7
npm install --save @nestjs/common@7.6.18 @nestjs/core@7.6.18 @nestjs/jwt@7.2.0 @nestjs/mongoose@7.2.4 @nestjs/passport@7.1.6 @nestjs/platform-express@7.6.18 @nestjs/schematics@7.3.1
npm install
ng update @nrwl/workspace@10.4.15
npm install
```
I tried updating to Nx 11:
```
nx migrate @nrwl/workspace@11.6.3
```
but that also failed (in @angular/cdk), so I manually updated again:
```
ng update @angular/cli@11.2.14
ng update @angular/material@11.2.13
ng update @nrwl/workspace@11.6.3
```
the last one failed part way through complaining that I used `ng update` instead of `nx migrate latest`. So I ran:
```
nx migrate latest
```
it failed complaining it couldn't find `tsconfig.base.json` and remembered that in one of the many previous attempts I had noticed that it had renamed `tsconfig,json`, so I did that... and the migration to 12 finally completed.

Now to deal with the resulting build errors.

**Comment by hfournier** [comment]
So I wasted a lot of time chasing down build errors that did not exist, because they were due to the fact that `nx migrate latest` had not updated `@angular/cdk`. After the migration, I found:
```
@angular/cli 12.2.7
@angular/core 12.2.0
@angular/material 12.2.0
@angular/cdk 11.2.13
```
why weren't they all updated to 12.2.7? The end result was that I was getting errors in my scss theme files and internally in the cdk.
To fix the problem, I tried running:
```
nx migrate @nrwl/workspace --to="@angular/material@12.2.0,@angular/cdk@12.2.0"
```
but it didn't do anything. It didn't even update the `package.json`.
So I had to disable Nx again by editing `package.json` and changing `"ng": "nx"` to `"ng": "ng"`, and deleting the postInstall line. I also changed `"@angular/material": "~12.2.0"` to `"@angular/material": "~11.2.13"`, so it would match the CDK and I could update them both. Then I deleted the `node_modules` folder and ran `npm install`. Finally, I was able to run:
```
ng update @angular/core
ng update @angular/material
```
which updated all things Angular to 12.2.7 and migrated the scss code properly and I was able to build without errors.
